tests/test_examples.py

```diff
@@ -43,3 +45,11 @@ class ExamplesTests(TestCase):
                            glob.glob(os.path.join(example_dir(), '*.py'))))
     def test_example(self, name, example):
         imp.load_source('__main__', os.path.basename(example), open(example))
+        # The algorithms buyapple.py and dual_moving_average.py have no
+        # if __name__ == '__main__' code block, and consequently are not
+        # executed by the imp.load_source() call. Instead, call these
+        # example algorithms directly as if using scripts/run_algo.py
+        if name == 'buyapple_py' or name == 'dual_moving_average_py':
+            confs = ['-f', example, '--start', '2011-1-1', '--end', '2012-1-1']
+            parsed_args = parse_args(confs)
+            run_pipeline(**parsed_args)
```

Comment:
@jfkirk chatted about this. James suggested adding `if __name__ == '__main__'` blocks into the two examples which are currently missing them. This has the advantage of being consistent across all algos and it allows the user to execute the algorithm directly using `python zipline/examples/<algo>.py` without the need for lots of command line flags. There are pros and cons to both idea. @twiecki have any thoughts?